File: two-pointers/intersection-of-two-linked-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
        ListNode currA = headA;
        ListNode currB = headB;
        if  (headA == null || headB == null) {
            return null;
        }
        else {
            while(currA != currB) {
                if (currA == null) {
                currA = headB;
                } else {
                    currA = currA.next;
                }

                if (currB == null) {
                    currB = headA;
                } else {
                    currB = currB.next;
                }
            }
        }
        return currA;
    }
}

File: hash-table/intersection-of-two-linked-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def getIntersectionNode(self, headA: ListNode, headB: ListNode) -> Optional[ListNode]:
        currA, currB = headA, headB
        while currA:
            currB = headB
            while currB:
                currB = currB.next
                if currA == currB:
                    return currA
            currA = currA.next
        return None
